pkg/reconciler/buildrun/buildrun.go

```diff
@@ -210,31 +209,33 @@ func (r *ReconcileBuildRun) Reconcile(ctx context.Context, request reconcile.Req
 
 			// make sure the BuildRun has not already been cancelled
 			if buildRun.IsCanceled() {
-				if updateErr := resources.UpdateConditionWithFalseStatus(ctx, r.client, buildRun, "the BuildRun is marked canceled.", buildv1alpha1.BuildRunStateCancel); updateErr != nil {
+				if updateErr := resources.UpdateConditionWithFalseStatus(ctx, r.client, buildRun, "the BuildRun is marked canceled.", buildv1beta1.BuildRunStateCancel); updateErr != nil {
 					return reconcile.Result{}, updateErr
 				}
 				return reconcile.Result{}, nil
 			}
 
-			// Set OwnerReference for Build and BuildRun only when build.shipwright.io/build-run-deletion is set "true"
-			if build.GetAnnotations()[buildv1alpha1.AnnotationBuildRunDeletion] == "true" && !resources.IsOwnedByBuild(build, buildRun.OwnerReferences) {
-				if err := r.setOwnerReferenceFunc(build, buildRun, r.scheme); err != nil {
-					build.Status.Reason = buildv1alpha1.BuildReasonPtr(buildv1alpha1.SetOwnerReferenceFailed)
-					build.Status.Message = pointer.String(fmt.Sprintf("unexpected error when trying to set the ownerreference: %v", err))
-					if err := r.client.Status().Update(ctx, build); err != nil {
-						return reconcile.Result{}, err
+			// Set OwnerReference for Build and BuildRun only when build retention AtBuildDeletion is set to "true"
```

Comment:
Do I see it correct that we never had code that removes the owner reference when AtBuildDeletion is changed to false?

EDIT: I just find it in pkg/validate/ownerreferences.go, hm, that is a weird location.